azkaban-common/src/test/java/azkaban/project/FlowTriggerTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import azkaban.Constants;
+import azkaban.utils.Props;
+import java.util.UUID;
+import org.junit.Test;
+
+
+public class FlowTriggerTest {
+
+  private Dependency createUniqueTestDependency(final String type) {
+    final UUID uuid = UUID.randomUUID();
+    return createTestDependency(type, uuid.toString());
+  }
+
+  private Dependency createTestDependency(final String type, final String name) {
+    final Props depProps = new Props();
+    depProps.put(Constants.DependencyProperties.DEPENDENCY_NAME, name);
+    depProps.put(Constants.DependencyProperties.DEPENDENCY_TYPE, type);
+    final Dependency dep = new Dependency(depProps);
+    return dep;
+  }
+
+  private FlowTrigger.FlowTriggerBuilder initTrigger() {
```

Comment:
Consider adding a test to validate schedule props?

Reply:
it hasn't been implemented yet. will add the unit test when ready.